KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,925 @@
+
+![layerDiagram](https://user-images.githubusercontent.com/32922423/85986440-355d0180-ba27-11ea-8475-afe7638e6ffb.png)
+
+The `KCT` layer allows you to interact with Klaytn compatible token (KCT) contracts (e.g. [KIP-7] or [KIP-17]). This is implemented by extending the `Contract` class in the `Contract` layer.
+
+The `Contract` layer allows you to interact with smart contracts on the Klaytn.
+
+The `ABI` layer provides the functions to encode and decode parameters based on ABI.
+
+The `Transaction` layer contains classes of Klaytn transaction types, `TransactionDecoder`, and `TransactionHasher`.
+
+The `TransactionDecoder` class decodes the RLP-encoded transaction string. The `TransactionHasher` class calculates the hash of the transaction.
+
+The `Wallet` layer contains keyring classes (`SingleKeyring`, `MultipleKeyring`, and `RoleBasedKeyring`) and `KeyringContainer` class that contains multiple keyrings. The `KeyringContainer` class acts as an "in-memory wallet" that stores the Keyring instances.
+
+The `Account` layer contains an `Account` class that stores information needed when updating the [AccountKey] of the account in the Klaytn.
+
+Also, the `RPC` layer includes the `Klay` class, which is responsible for rpc calls of the klay namespace, and the `Net` class, which is responsible for rpc calls of the net namespace.
+
+Finally, the `Utils` layer contains the `Utils` class that provides utility functions.
+
+From the next chapter, each layer is described in detail.
+
+### Account Layer Class Diagram
+
+The `Account` layer provides functionality related to updating the [AccountKey] of the Klaytn account.
+
+![finalAccount](https://user-images.githubusercontent.com/32922423/89011409-44e3a900-d34b-11ea-8dc8-bed423c4e17a.png)
+
+`Account` is a class that contains information needed to update the [AccountKey] of the account in the Klaytn. It has `address` and `accountKey` as member variables. The `accountKey` can be an instance of (`AccountKeyLegacy`, `AccountKeyPublic`, `AccountKeyFail`, `AccountKeyWeightedMultiSig` or `AccountKeyRoleBased`) depending on the key.
+
+An `Account` instance is injected into the `account` field of the AccountUpdate transaction types (`AccountUpdate`, `FeeDelegatedAccountUpdate` or `FeeDelegatedAccountUpdateWithRatio`). When an AccountUpdate transaction is successfully processed in Klaytn, the account key of the [Klaytn account] is updated with the `accountKey` defined in the `Account` instance.
+
+Interface for AccountKey is defined as `IAccountKey`, and account key classes (`AccountKeyLegacy`, `AccountKeyPublic`, `AccountKeyFail`, `AccountKeyWeightedMultiSig` and `AccountKeyRoleBased`) implements IAccountKey.
+
+`WeightedMultiSigOptions` is provided for `AccountKeyWeightedMultiSig` where the threshold and the weight of each key are defined.
+
+The `AccountKeyDecoder` class decodes the RLP-encoded string using the decode function implemented in each AccountKey class.
+
+#### IAccountKey
```

Comment:
@jimni1222 Need descriptions for all classes before explaining their methods.

Reply:
Hm... For the purpose of describing methods, I've only described the classes in which the methods exist here.
If you want a description of classes, I think that classes that don't have methods should also be listed.

Reply:
It would be better to include all classes with their attributes. Sorry for the endless work. T_T

Reply:
@kjhman21 I added descriptions for class and member variables. Please review this when you have time :) Thank you!